automated/linux/kselftest/kselftest.sh

```diff
@@ -110,15 +110,42 @@ fi
 
 
 parse_output() {
-    grep "^selftests:" "${LOGFILE}" > "${RESULT_FILE}"
+    perl -ne '
+    if (m|^# selftests: (.*)$|) {
+	$testdir = $1;
+	$testdir =~ s|[:/]\s*|.|g;
+    } elsif (m|^(?:# )*(not )?ok (?:\d+) ([^#]+)(# (SKIP)?)?|) {
+        $not = $1;
+        $test = $2;
+        $skip = $4;
+        $test =~ s|\s+$||;
+        # If the test name starts with "selftests: " it is "fully qualified".
+        if ($test =~ /selftests: (.*)/) {
+            $test = $1;
+	    $test =~ s|[:/]\s*|.|g;
+        } else {
+            # Otherwise, it likely needs the testdir prepended.
+            $test = "$testdir.$test";
+        }
+        # Any appearance of the SKIP is a skip.
+        if ($skip eq "SKIP") {
+            $result="skip";
+        } elsif ($not eq "not ") {
+            $result="fail";
+        } else {
+            $result="pass";
+        }
+	print "$test $result\n";
```

Comment:
with your patch applied i see great improvement in test count increased from 375 to 934 in total.

Before patch applied:
PASS    FAIL   TOTAL
173      168     375



After patch applied:
PASS  FAIL   TOTAL
702     186    934
https://lkft.validation.linaro.org/results/1766015